_query-dsl/query-dsl/compound/function-score.md

```diff
@@ -0,0 +1,827 @@
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 2,
+      "relation": "eq"
+    },
+    "max_score": 1,
+    "hits": [
+      {
+        "_index": "hotels",
+        "_id": "1",
+        "_score": 1,
+        "_source": {
+          "name": "Hotel Within 200",
+          "location": {
+            "lat": 40.7105,
+            "lon": 74
+          }
+        }
+      },
+      {
+        "_index": "hotels",
+        "_id": "2",
+        "_score": 0.20099315,
+        "_source": {
+          "name": "Hotel Outside 500",
+          "location": {
+            "lat": 40.7115,
+            "lon": 74
+          }
+        }
+      }
+    ]
+  }
+}
+```
+</details>
+
+### Parameters
+
+The following table lists all parameters supported by the `gauss`, `exp`, and `linear` functions.
+
+Parameter | Description
+:--- | :---
+`origin` | The point from which to calculate the distance. Must be provided as a number for numeric fields, date for date fields, and geopoint for geopoint fields. Required for geopoint and numeric fields. Optional for date fields (defaults to `now`). For date fields, date math is supported (for example, `now-2d`).
+`offset` | Defines the distance from the origin within which documents are given a score of 1. Optional. Default is 0.
+`scale` | Documents at the distance of `scale` + `offset` from the `origin` are assigned a score of `decay`. Required. <br>For numeric fields, `scale` can be any number. <br>For date fields, `scale` can be defined as a number and [units]({{site.url}}{{site.baseurl}}/api-reference/units/) (`5h`, `1d`). If units are not provided, defaults to milliseconds. <br>For geopoint fields, `scale` can be defined as a number and [units]({{site.url}}{{site.baseurl}}/api-reference/units/) (`1mi`, `5km`). If units are not provided, defaults to meters.
+`decay` | Defines the score of a document at the distance of `scale` + `offset` from the `origin`. Optional. Default is 0.5.
+
+For fields that are missing from the document, decay functions return a score of 1.
+{: .note}
+
+### Example: Numeric fields
+
+The following query uses the exponential decay function to prioritize blogs by the number of comments:
+
+```json
+GET blogs/_search
+{
+  "query": {
+    "function_score": {
+      "functions": [
+        {
+          "exp": {
+            "comments": { 
+              "origin":  "20",
+              "offset": "5",
+              "scale":  "10"
+            }
+          }
+        }
+      ]
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+The first two blogs in the results have a score of 1 because one is at the origin (20) and the other is at a distance 16 which is within the offset (the range at which documents receive a full score is calculated as 20 $$\pm$$ 5 and is [15, 25]). The third blog is at a distance of `scale` + `offset` from the `origin` (20 &minus; (5 + 10) = 15), so it is given the default `decay` score (0.5):
```

Comment:
```suggestion
The first two blogs in the results have a score of 1 because one is at the origin (20) and the other is at a distance 16, which is within the offset (the range at which documents receive a full score is calculated as 20 $$\pm$$ 5 and is [15, 25]). The third blog is at a distance of `scale` + `offset` from the `origin` (20 &minus; (5 + 10) = 15), so it is given the default `decay` score (0.5):
```